mindsdb/utilities/config.py

```diff
@@ -1,41 +1,57 @@
 import os
 import json
-
+import hashlib
 
 class Config(object):
     _config = {}
 
-    def __init__(self, config='mindsdb/default_config.json'):
-        if isinstance(config, dict):
-            self._update_recursive(self._config, config)
-        elif isinstance(config, str):
-            self.merge(config)
+    def __init__(self, config_path):
+        self._config_path = None
+        self._config_hash = None
+        self._config = None
+        if isinstance(config_path, str):
+            self.config_path = config_path
+            self._read()
+            self._config_hash = self._gen_hash()
+        else:
+            raise TypeError('Argument must be string representing a file path <Later on to be switched to file path and/or database connection info>')
+
+    def _read(self):
+        if isinstance(self.config_path, str) and os.path.isfile(self.config_path):
+            with open(self.config_path, 'r') as fp:
+                self._config = config = json.load(fp)
         else:
-            raise TypeError('argument must be string or dict')
-
-    def merge(self, config):
-        if isinstance(config, dict):
-            self._update_recursive(self._config, config)
-        elif isinstance(config, str) and os.path.isfile(config):
-            with open(config, 'r') as fp:
-                config = json.load(fp)
-            self._update_recursive(self._config, config)
-
-    def _update_recursive(self, a, b):
-        for key in [x for x in a.keys() if x in b]:
-            if isinstance(a[key], dict) and isinstance(b[key], dict):
-                self._update_recursive(a[key], b[key])
-            else:
-                a[key] = b[key]
-        for key in b.keys():
-            if key not in a:
-                a[key] = b[key]
+            raise TypeError('`self.config_path` must be a string representing a local file path to a json config')
+
+    def _gen_hash(self):
+        with open(self.config_path, 'rb') as fp:
+            return hashlib.md5(fp.read()).hexdigest()
+
+    def _set_updated(self, key):
+        # Only check this for dynamically generated keys, won't be needed once we switch to using a database here
+        if key in self._config['integrations']:
+            file_hash = self._gen_hash()
+            if file_hash != self._config_hash:
+                self._read()
+                self._config_hash = self._gen_hash()
 
     def __getitem__(self, key):
+        self._set_updated(key)
         return self._config[key]
 
     def get(self, key, default=None):
+        self._set_updated(key)
```

Comment:
this never will works, because 'key' is always top level keys of config (api, integration, interface, etc) and it never will be in self._config['integrations']

Reply:
My bad that was meant to read just `['integrations']`